app/forms.py

```diff
@@ -15,3 +17,42 @@ def __init__(self, *args, **kwargs):
         super(ProfileUpdateForm, self).__init__(*args, **kwargs)
         self.fields["tags"].widget = CheckboxSelectMultiple()
         self.fields["tags"].queryset = Tag.objects.all()
+
+
+class SocialLinkForm(ModelForm):
+    class Meta:
+        model = SocialLinkAttachement
+        fields = "__all__"
+
+    def __init__(self, *args, **kwargs):
+        if "obj" in kwargs:
+            self.object = kwargs.pop("obj")
+
+        super(SocialLinkForm, self).__init__(*args, **kwargs)
+        self.fields["content_type"].widget = HiddenInput()
+        self.fields["object_id"].widget = HiddenInput()
+
+    def clean_object_id(self):
+        cleaned_data = self.cleaned_data
+        if getattr(self, "object", None) is not None:
+            cleaned_data["object_id"] = self.object.pk
+        elif self.fields["object_id"].initial is not None:
+            cleaned_data["object_id"] = self.fields["object_id"].initial
+        return cleaned_data["object_id"]
+
+    def clean_content_type(self):
+        cleaned_data = self.cleaned_data
+        if "content_type" not in cleaned_data or cleaned_data[
+            "content_type"
+        ] != ContentType.objects.get_for_model(self.object.__class__):
+            raise ValidationError("content_type should not be modified")
+        return cleaned_data["content_type"]
```

Comment:
Relevant to the above; on second thought, consistency would be better. Should errors be thrown if the user manually edits these fields or should they be quietly reassigned?

Reply:
Ditto. In general, we don't tell the user they did something they weren't supposed to do.